### PR TITLE
Update registry code from serviceclass to clusterserviceclass

### PR DESCRIPTION
**Pull request by eriknelson**
This PR updates the registry code and file organization to correctly categorize what was formerly "ServiceClass" to "ClusterServiceClass". This makes way to introduce the namespaced resource "ServiceClass" registry code. This PR is a functional NOOP.

### CODE REVIEW

File: pkg/registry/servicecatalog/clusterserviceclass/strategy.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2017 The Kubernetes Authors.
```
Comment (n3wscott) [inline pkg/registry/servicecatalog/clusterserviceclass/strategy.go]: This date needs to be updated to 2018

File: pkg/registry/servicecatalog/clusterserviceclass/strategy.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+// RESTUpdateStrategy. This implementation validates updates to
+// clusterServiceClass.Status updates only and disallows any modifications to
+// the clusterServiceClass.Spec.
+type clusterServiceClassStatusRESTStrategy struct {
+	clusterServiceClassRESTStrategy
+}
+
+var (
+	clusterServiceClassRESTStrategies = clusterServiceClassRESTStrategy{
+		// embeds to pull in existing code behavior from upstream
```
Comment (n3wscott) [inline pkg/registry/servicecatalog/clusterserviceclass/strategy.go]: What is this comment tied to?

File: pkg/registry/servicecatalog/clusterserviceclass/strategy.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+func (clusterServiceClassRESTStrategy) Canonicalize(obj runtime.Object) {
+	_, ok := obj.(*sc.ClusterServiceClass)
+	if !ok {
+		glog.Fatal("received a non-clusterserviceclass object to create")
+	}
+}
+
+// NamespaceScoped returns false as ClusterServiceClass are not scoped to a
+// namespace.
+func (clusterServiceClassRESTStrategy) NamespaceScoped() bool {
```
Comment (n3wscott) [inline pkg/registry/servicecatalog/clusterserviceclass/strategy.go]: I don't know what the rest of k8s does, but maybe this is called just `Namespaced` ?

Comment (n3wscott) [reply pkg/registry/servicecatalog/clusterserviceclass/strategy.go]: 🤷‍♂️

File: pkg/registry/servicecatalog/clusterserviceclass/strategy.go (hunk 2 above)
Comment (eriknelson) [reply pkg/registry/servicecatalog/clusterserviceclass/strategy.go]: :man_shrugging: This is pulled verbatim from the existing code.

**Comment by eriknelson** [comment]
@carolynvs +1, I wanted to keep this one isolated on the rename without messing about with existing code to avoid accidentally introducing something unexpected.

**Comment by n3wscott** [comment]
LGTM I did not realize it was just a move. Thanks for clarifying.

**Pull request merged**